src/sql/workbench/browser/modal/modal.ts

```diff
@@ -168,40 +189,76 @@ export abstract class Modal extends Disposable implements IThemable {
 
 	/**
 	 * Build and render the modal, will call {@link Modal#renderBody}
+	 *
 	 */
 	public render() {
+		let hasMacClass: boolean = document.querySelector('body.mac') !== null ? true : false;
 		let builderClass = '.modal.fade';
-		if (this._modalOptions.isFlyout) {
-			builderClass += '.flyout-dialog';
-		}
+		builderClass += this._modalOptions.dialogStyle === 'flyout' ? '.flyout-dialog'
+			: this._modalOptions.dialogStyle === 'callout' ? '.callout-dialog'
+				: this._modalOptions.dialogStyle === 'calloutCompact' ? '.callout-dialog.compact'
+					: '';
 
 		this._bodyContainer = DOM.$(`${builderClass}`, { role: 'dialog', 'aria-label': this._title });
 		const top = this.layoutService.offset?.top ?? 0;
 		this._bodyContainer.style.top = `${top}px`;
 		this._modalDialog = DOM.append(this._bodyContainer, DOM.$('.modal-dialog'));
-		this._modalContent = DOM.append(this._modalDialog, DOM.$('.modal-content'));
+
+		if (this._modalOptions.dialogStyle === 'callout' || this._modalOptions.dialogStyle === 'calloutCompact') {
+			let arrowClass = `.arrow.from-${this._modalOptions.dialogPosition}`;
+			this._modalContent = DOM.append(this._modalDialog, DOM.$(`.modal-content${arrowClass}`));
+
+			if (this._modalOptions.dialogPosition === 'below') {
+				if (hasMacClass) {
```

Comment:
Explain why this is needed in a comment please

Reply:
@Charles-Gagnon I don't understand. Do you want me to clarify this section with a comment?

Reply:
Yes.

In general though I need help understanding myself why we need to do this at all - what is it about macs that make you need to do this special offset? And then especially needing to do it from the extension side too? How are extension authors supposed to know that this is required in order to have it displayed correctly? Is there anything we can do to remove all this specialized mac stuff being necessary?

Reply:
I removed all dependencies upon a CSS class on the body tag. As stated in a prior comment, there is now a 30px difference in vertical positioning of callout on Mac and Windows.